src/Explorer/Tabs/DocumentsTabV2/DocumentsTabStateUtil.ts

```diff
@@ -0,0 +1,96 @@
+// Definitions of State data
+
+import { deleteState, loadState, saveState, saveStateDebounced } from "Shared/AppStatePersistenceUtility";
+import { userContext } from "UserContext";
+import * as ViewModels from "../../../Contracts/ViewModels";
+import { Action } from "../../../Shared/Telemetry/TelemetryConstants";
+import * as TelemetryProcessor from "../../../Shared/Telemetry/TelemetryProcessor";
+
+const componentName = "DocumentsTab";
+export type SubComponentName = "ColumnSizes" | "FilterHistory" | "MainTabDivider";
```

Comment:
Maybe use string enums for these?

Reply:
I like union type for its simplicity: it's a simple check at compile time, but enum could be useful to enumerate all values.